programs/loader-v4/src/lib.rs

```diff
@@ -328,26 +322,23 @@ pub fn process_instruction_truncate(
     };
     match program.get_lamports().cmp(&required_lamports) {
         std::cmp::Ordering::Less => {
-            let lamports_to_pay = required_lamports.saturating_sub(program.get_lamports());
-            if !payer_or_recipient.is_signer() {
-                ic_logger_msg!(log_collector, "Payer did not sign");
-                return Err(InstructionError::MissingRequiredSignature);
-            }
-            if payer_or_recipient.get_lamports() < lamports_to_pay {
-                ic_logger_msg!(
-                    log_collector,
-                    "Insufficient lamports, {} are required",
-                    required_lamports
-                );
-                return Err(InstructionError::InsufficientFunds);
-            }
-            payer_or_recipient.checked_sub_lamports(lamports_to_pay)?;
```

Comment:
I am likely confused, but my understanding was that the payer will pay the lamports if more lamports were needed as part of this instruction. The logic to withdraw lamports from the program account will be moved out. Is that not correct?

Reply:
The other way, the logic to top up accounts is removed, because one can increase the lamports without being the owner anyway, so it is simply redundant. The logic to withdraw superfluous funds is needed as only the loader program can decrease the lamports.